ui/src/app/domains/views/DomainDetails/ResourceRecords/EditRecordDomainForm.tsx

```diff
@@ -0,0 +1,137 @@
+
+import AddRecordFields from "../DomainDetailsHeader/AddRecordDomainForm/AddRecordFields";
+
+import FormikForm from "app/base/components/FormikForm";
+import { actions as domainActions } from "app/store/domain";
+import domainSelectors from "app/store/domain/selectors";
+import type {
+  UpdateResourceParams,
+  UpdateAddressRecordParams,
+} from "app/store/domain/types";
+import type { Domain, DomainResource } from "app/store/domain/types/base";
+import { RecordType } from "app/store/domain/types/base";
+
+type Props = {
+  id: Domain["id"];
+  resource: DomainResource;
+  closeForm: () => void;
+};
+
+export type CreateRecordValues = {
+  name: DomainResource["name"];
+  rrtype: DomainResource["rrtype"];
+  rrdata: DomainResource["rrdata"];
+  ttl: DomainResource["ttl"] | "";
+};
+
+const EditRecordDomainForm = ({
+  id,
+  resource,
+  closeForm,
+}: Props): JSX.Element => {
+  const dispatch = useDispatch();
+  const errors = useSelector(domainSelectors.errors);
+  const saved = useSelector(domainSelectors.saved);
+  const saving = useSelector(domainSelectors.saving);
+  const cleanup = () => dispatch(domainActions.cleanup());
+
+  const CreateRecordSchema: SchemaOf<CreateRecordValues> = Yup.object()
+    .shape({
+      name: Yup.string().required("This field is required name"),
+      rrtype: Yup.string().required("This field is required tuype"),
+      rrdata: Yup.string().required("This field is required data"),
+      ttl: Yup.number()
+        .nullable(true)
+        .min(0, "Ensure this value is greater than or equal to 0."),
+    })
+    .defined();
+
+  const updateAddressRecord = (values: CreateRecordValues) => {
+    const params: UpdateAddressRecordParams = {
+      ...resource,
+      domain: id,
+      previous_name: resource["name"],
+      previous_rrdata: resource["rrdata"],
+      previous_rrtype: resource["rrtype"],
+      previous_ttl: resource["ttl"],
+      name: values.name,
+      rrdata: values.rrdata,
+      rrtype: values.rrtype,
+      address_ttl:
+        values.ttl === "" || values.ttl === null ? null : Number(values.ttl),
+      ip_addresses: (values.rrdata ?? "").split(/[ ,]+/),
+    };
+
+    if (values.name !== resource.name) {
+      dispatch(domainActions.updateDnsResource(params));
+    }
+
+    dispatch(domainActions.updateAddressRecord(params));
+  };
+
+  const updateDnsRecord = (values: CreateRecordValues) => {
+    const params: UpdateResourceParams = {
+      ...resource,
+      domain: id,
+      previous_name: resource["name"],
+      previous_rrdata: resource["rrdata"],
+      previous_rrtype: resource["rrtype"],
+      previous_ttl: resource["ttl"],
+      name: values.name,
+      rrdata: values.rrdata,
+      rrtype: values.rrtype,
+      ttl: values.ttl === "" || values.ttl === null ? null : Number(values.ttl),
+    };
+
+    if (values.name !== resource.name) {
+      dispatch(domainActions.updateDnsResource(params));
+    }
+
+    dispatch(domainActions.updateDnsData(params));
```

Comment:
@huwshimi @Caleb-Ellis  In here we dispatch updateDnsResource and updateDnsData immediately, but legacy code calls them in sequence of promises. This is mentioned in canonical-web-and-design/app-squad#113